src/mapbox/layers/TralisLayer.js

```diff
@@ -95,20 +97,38 @@ class TralisLayer extends mixin(TrackerLayer) {
       this.api
         .getFullTrajectory(id, this.mode, this.generalizationLevel)
         .then((fullTrajectory) => {
-          const stroke = fullTrajectory.features[0].properties.stroke
-          if (stroke && stroke[0] !== '#') {
-            fullTrajectory.features[0].properties.stroke = `#${stroke}`;
+           const type = fullTrajectory.features[0].properties.type
+           fullTrajectory.features[0].properties.typeIdx = getTypeIndex(type)
+
+          let lineColor = fullTrajectory.features[0].properties.stroke
+          if (lineColor && lineColor[0] !== '#') {
+            lineColor = `#${lineColor}`
+            fullTrajectory.features[0].properties.stroke = lineColor;
+          }
+          else if(!lineColor){
+              lineColor = bgColors[types.findIndex((t) => t.test(type))]
           }
-          const type = fullTrajectory.features[0].properties.type
-          fullTrajectory.features[0].properties.typeIdx = getTypeIndex(type)
+
           fullTrajectory.features[0].geometry.geometries.forEach(element => {
             const newCoords = []
             for (const coord of element.coordinates) {
               newCoords.push(toLonLat(coord))
             }
             element.coordinates = newCoords
           });
+
+          const linePaintInterpolation = [
+              'interpolate',
+              ['linear'],
+              ['line-progress'],
+              0,
+              '#BEBEBE',
+              1,
+              lineColor
+          ]
```

Comment:
It would be better to use cubic bezier feature to display this.
Example from tomtom routes:
`['interpolate', ['cubic-bezier', 1, 0, 1, 0], ['line-progress'], 0, 'green']`

Reply:
Thanks, I will change it.